lecture2/code.py

```diff
@@ -0,0 +1,133 @@
+#######################################################
+"""Personal Tests Below This rectanle! """
+#######################################################
+"""
+def concatinate_strings(string1, string2):
+    string1 = "Hello"
+    string2 = "World"
+    string_concat = ""
+    if string_concat != "HelloWorld":
+        return string1 + string2
+    else:
+        return string1, string2 and string_concat            
+
+2nd attempt at using the not in check in Python
+
+def concatinate_strings(string1, string2):
+    string1 = "Hello"
+    string2 = "World"
+    string_concat = ""
+    if "HelloWorld" not in string_concat:
+        string_concat = string1 + string2
+        return string_concat
+    if "Hello" not in string2:
+        return string1
+    if "World" not in string1:
+        return string2          
+    else:
+        return ""
+
+"""
+#######################################################
+# Regular test cases below this rectangle.
+#######################################################
+def concatinate_strings(string1, string2):
+    string_concat = string1 + string2
+    #"%s%s".format(string1, string2)
+    #f"{string1}{string2}"
+    # #string1 + string2
+    return string_concat
+# read up on string.join() & f.string & string formatting.
+
+
+def prefix_string(string, prefix):
+    # example of string prefix 
+    # new_string = r"Hello, World" - outputs raw string literal.
+    # new_string_as_byte = b"Hello, World" - outputs bytes string literal.
+    return prefix + string
+
+
+def suffix_string(string, suffix):
+    """
+        example if string ends with a specified suffix: "Hello World"; 
+        World = suffix, then return True
+        else: return False
+    """
+    return string + suffix
+
+
+def starts_with(string, prefix):
+    """
+        test prefix: Hello World -> Starts with He -> return True
+        if test prefix Hello World !-> He -> return False
+        if Hello = Hello -> return True
+        if Hello World = Hello -> return False
+
+    """
+    if string.startswith(prefix):
+        return True
+    else:
+        return False    
+
+
+def ends_with(string, suffix):
+    """
+        using str.endswith() to test if string "Hello World" ends with "World:
+            if "Hello World".endswith("World"):
+                return True
+            else:
+                return False    
+    """
+    if string.endswith(suffix):
+        return True
+    else:
+        return False    
+
+
+def get_first_character(string):
+    """
+        return the index of a given string: "Hello World" expected output: str[0] = H
```

Comment:
please be correct in your comments. Index points to any position in string. Index = points to first char since indexing in array (string is array of chars) in python is 0-based (starts at 0)